src/main/java/io/sqreen/powerwaf/PowerwafContext.java

```diff
@@ -154,9 +154,7 @@ public Powerwaf.ResultWithData runRules(Map<String, Object> parameters,
                 if (metrics != null) {
                     long after = System.nanoTime();
                     long totalTimeNs = after - before;
-                    synchronized (metrics) {
-                        metrics.totalRunTimeNs += totalTimeNs;
-                    }
+                    metrics.addTotalRunTimeNs(totalTimeNs);
```

Comment:
If we are increasing the time here, and also within the C code called here (I haven't double checked), then are we double counting the time?